docs/contributing/component-js-configuration-options.md

```diff
@@ -0,0 +1,192 @@
+# Adding JavaScript configuration options to components
+
+Configuration options permit developers to pass additional information into a component's JavaScript, allowing them to change how the component might work or appear without needing to create their own fork.
+
+Configuration options straddle the line between Nunjucks, HTML and JavaScript.
```

Comment:
What does 'straddle the line' mean in this context? Is it the Nunjucks, HTML and JavaScript all use the same configuration options or that they can recognise each others configuration options or something similar?

Reply:
I guess what I meant here is that—although configuration options are only currently used for JavaScript—the manner in which they are set exists across all three of these contexts.

I was trying to set up the next few sections, basically 😅

Reply:
ok. So could we use your sentence something like this?
Currently, only JavaScript uses the configuration options, but due to the way they are written Nunjucks and HTML can use them as well.
Is that accurate?

Reply:
It's more like the options can be configured using Nunjucks, HTML or JavaScript, but only JavaScript will use them?

Reply:
Oh I see. I had it almost totally backwards! Yes, that sounds good.